File: algorithms/303. Range Sum Query - Immutable/solution.py
class NumArray(object):

    def __init__(self, nums):
        """
        :type nums: List[int]
        """
        self.sums = [0]
        for n in nums:
            self.sums.append(self.sums[-1] + n)

    def sumRange(self, i, j):
        """
        :type i: int
        :type j: int
        :rtype: int
        """
        return self.sums[j + 1] - self.sums[i]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)

File: algorithms/303. Range Sum Query - Immutable/solution.cc
class NumArray {
    vector<int> sums;
public:
    NumArray(vector<int> nums) {
        sums.push_back(0);
        for ( int n : nums ) {
            sums.push_back(sums.back() + n);
        }
    }
    
    int sumRange(int i, int j) {
        return sums[j + 1] - sums[i];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(i,j);
 */
